src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -1263,24 +1243,61 @@ internal override void ForceComplete(SourceLocation locationOpt, CancellationTok
                                     parameter.Type.CheckAllConstraints(conversions, parameter.Locations[0], diagnostics);
                                 }
 
-                                if (_state.NotePartComplete(CompletionPart.Parameters))
+                                if (_state.NotePartComplete(CompletionPart.StartPropertyParameters))
```

Comment:
Should `if (_state.NotePartComplete(CompletionPart.StartPropertyParameters)) { ... }` wrap the entire `case`, so that only one thread needs to check parameter constraints?